scripts/operations/system_layout_service/add_cdu_switch.py

```diff
@@ -36,32 +37,87 @@ def find_subnet(sls_network, name):
         if subnet["Name"] == name:
             network_hardware_subnet = subnet
             break
-    
+
     return  network_hardware_subnet
 
-def find_next_available_ip(sls_subnet):
+def find_next_available_ip(sls_subnet, reserved_ips=[]):
     subnet = netaddr.IPNetwork(sls_subnet["CIDR"])
 
     existing_ip_reservations = netaddr.IPSet()
-    existing_ip_reservations.add(sls_subnet["Gateway"])
+
     for ip_reservation in sls_subnet["IPReservations"]:
         print("  Found existing IP reservation {} with IP {}".format(ip_reservation["Name"], ip_reservation["IPAddress"]))
         existing_ip_reservations.add(ip_reservation["IPAddress"])
 
-    for available_ip in list(subnet[1:-2]):
+    start_ip = None
+    if len(ip_reservation) > 0:
```

Comment:
I'm still very new at python so using code reviews as a way to learn.  :)  Wondering why this is len(ip_reservation) instead of len(existing_ip_reservations).   If there was any match in the subnet list I can see that both ip_reservation and existing_ip_reservations would be > 0.  Just curious if there's a deeper reason to chose ip_reservation, or if it was just personal choice.  I'm ok either way.

Reply:
That's a good catch. It should be len(existing_ip_reservations)